src/main/java/com/epam/test_generator/services/TestResultService.java

```diff
@@ -0,0 +1,254 @@
+package com.epam.test_generator.services;
+
+import com.epam.test_generator.config.security.AuthenticatedUser;
+import com.epam.test_generator.dao.interfaces.TestResultDAO;
+import com.epam.test_generator.dto.RawCaseResultDTO;
+import com.epam.test_generator.dto.RawStepResultDTO;
+import com.epam.test_generator.dto.RawSuitResultDTO;
+import com.epam.test_generator.dto.TestResultDTO;
+import com.epam.test_generator.entities.Case;
+import com.epam.test_generator.entities.CaseResult;
+import com.epam.test_generator.entities.Project;
+import com.epam.test_generator.entities.Status;
+import com.epam.test_generator.entities.StepResult;
+import com.epam.test_generator.entities.Suit;
+import com.epam.test_generator.entities.SuitResult;
+import com.epam.test_generator.entities.TestResult;
+import com.epam.test_generator.services.exceptions.BadRequestException;
+import com.epam.test_generator.transformers.TestResultTransformer;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.transaction.Transactional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+@Transactional
+@Service
+public class TestResultService {
+
+    @Autowired
+    private TestResultDAO testResultDAO;
+
+    @Autowired
+    private SuitService suitService;
+
+    @Autowired
+    private CaseService caseService;
+
+    @Autowired
+    private StepService stepService;
+
+    @Autowired
+    private ProjectService projectService;
+
+    @Autowired
+    private TestResultTransformer testResultTransformer;
+
+    /**
+     * Parse all crucial information from List of {@link RawSuitResultDTO}
+     * and collect it into {@link TestResult} struct class.
+     *
+     * @param projectId identifier of {@link Project}
+     * @param suitResultDTOS List of {@link RawSuitResultDTO}.
+     * @param authentication current user principals
+     * @return {@link TestResult}
+     */
+    public TestResult saveResult(Long projectId,
```

Comment:
It is a really huge method. You can extract some methods. For example you can extract inner loops and maybe something else.